plugins/modules/device_credential_workflow_manager.py

```diff
@@ -1192,416 +1174,416 @@ def get_snmpV3_params(self, snmpV3Details):
 
         Parameters:
-            CredentialDetails (dict) - Playbook details containing Global Device Credentials.
+            credential_details (dict) - Playbook details containing Global Device Credentials.
             global_credentials (dict) - All global device credentials details.
 
         Returns:
-            snmpV3Details (List) - The current snmp_v3.
+            snmp_v3_details (List) - The current snmp_v3.
         """
 
         # Playbook snmp_v3 Credential details
-        all_snmpV3 = CredentialDetails.get("snmp_v3")
+        all_snmp_v3 = credential_details.get("snmp_v3")
 
         # All snmp_v3 details from the Cisco Catalyst Center
-        snmpV3_details = global_credentials.get("snmpV3")
+        snmp_v3__details = global_credentials.get("snmpV3")
 
         # Cisco Catalyst Center details for the snmp_v3 Credential given in the playbook
-        snmpV3Details = []
-        if all_snmpV3 and snmpV3_details:
-            for snmpV3Credential in all_snmpV3:
-                snmpV3Detail = None
-                snmpV3Id = snmpV3Credential.get("id")
-                if snmpV3Id:
-                    snmpV3Detail = get_dict_result(snmpV3_details, "id", snmpV3Id)
-                    if not snmpV3Detail:
+        snmp_v3_details = []
+        if all_snmp_v3 and snmp_v3__details:
+            for snmp_v3_credential in all_snmp_v3:
+                snmp_v3_detail = None
+                snmp_v3_id = snmp_v3_credential.get("id")
+                if snmp_v3_id:
+                    snmp_v3_detail = get_dict_result(snmp_v3__details, "id", snmp_v3_id)
+                    if not snmp_v3_detail:
                         self.msg = "snmp_v3 credential id is invalid"
                         self.status = "failed"
                         return self.check_return_status()
 
-                snmpV3Description = snmpV3Credential.get("description")
-                if snmpV3Description and (not snmpV3Detail):
-                    snmpV3Detail = get_dict_result(snmpV3_details, "description", snmpV3Description)
+                snmp_v3_description = snmp_v3_credential.get("description")
+                if snmp_v3_description and (not snmp_v3_detail):
+                    snmp_v3_detail = get_dict_result(snmp_v3__details, "description", snmp_v3_description)
 
-                if not snmpV3Detail:
-                    snmpV3OldDescription = snmpV3Credential.get("old_description")
-                    if snmpV3OldDescription and (not snmpV3Detail):
-                        snmpV3Detail = get_dict_result(snmpV3_details, "description", snmpV3OldDescription)
-                        if not snmpV3Detail:
+                if not snmp_v3_detail:
+                    snmp_v3_old_description = snmp_v3_credential.get("old_description")
+                    if snmp_v3_old_description and (not snmp_v3_detail):
+                        snmp_v3_detail = get_dict_result(snmp_v3__details, "description", snmp_v3_old_description)
```

Comment:
Have you tested your code? 

There are double underscores ... snmp_**v3__details**

Reply:
yes, code has been tested.
changed snmp_v3__details as snmpv3_details